src/gui/creds/httpcredentialsgui.cpp

```diff
@@ -113,39 +115,15 @@ void HttpCredentialsGui::asyncAuthResult(OAuth::Result r, const QString &user,
 
 void HttpCredentialsGui::showDialog()
 {
-    QString msg = tr("Please enter %1 password:<br>"
-                     "<br>"
-                     "User: %2<br>"
-                     "Account: %3<br>")
-                      .arg(Utility::escape(Theme::instance()->appNameGUI()),
-                          Utility::escape(_user),
-                          Utility::escape(_account->displayName()));
-
-    QString reqTxt = requestAppPasswordText(_account);
-    if (!reqTxt.isEmpty()) {
-        msg += QLatin1String("<br>") + reqTxt + QLatin1String("<br>");
-    }
-    if (!_fetchErrorString.isEmpty()) {
-        msg += QLatin1String("<br>")
-            + tr("Reading from keychain failed with error: '%1'")
-                  .arg(Utility::escape(_fetchErrorString))
-            + QLatin1String("<br>");
-    }
+    auto *contentWidget = new HttpLoginRequiredWidget(_account->sharedFromThis());
+    auto *dialog = new LoginRequiredDialog(contentWidget, ocApp()->gui()->settingsDialog());
 
-    QInputDialog *dialog = new QInputDialog(ocApp()->gui()->settingsDialog());
-    dialog->setAttribute(Qt::WA_DeleteOnClose, true);
-    dialog->setWindowTitle(tr("Enter Password"));
-    dialog->setLabelText(msg);
-    dialog->setTextValue(_previousPassword);
-    dialog->setTextEchoMode(QLineEdit::Password);
-    if (QLabel *dialogLabel = dialog->findChild<QLabel *>()) {
-        dialogLabel->setOpenExternalLinks(true);
-        dialogLabel->setTextFormat(Qt::RichText);
-    }
+    connect(dialog, &LoginRequiredDialog::finished, ocApp()->gui()->settingsDialog(), [this, dialog, contentWidget](const int result) {
+        contentWidget->deleteLater();
```

Comment:
contentWidget should be handled by parenting?

Reply:
Right now the content widget has to be passed to the constructor (which makes sense, since such an object is required for the parent to work). Therefore, I have two... workarounds in mind:

- call `contentWidget->setParent` after setting up the parent
- use a setter to pass `contentWidget` to the dialog

Reply:
The parent of contentWidget will already be the LoginRequiredDialog as the widget will be added to one of its layouts.
Widgets take the ownership of their children.

Reply:
Nevermind then.